### PR TITLE
Visual feedback on invalid input telerik/kendo#6759

### PR DESCRIPTION
**Pull request by dbalinov**
telerik/kendo#6759

Test file: [test.zip](https://github.com/telerik/kendo-ui-core/files/850626/test.zip)

### CODE REVIEW

File: src/kendo.numerictextbox.js [hunk 1]
```diff
@@ -308,6 +310,16 @@ var __meta__ = { // jshint ignore:line
             that._downArrowEventHandler = new kendo.UserEvents(that._downArrow, { release: _release });
         },
 
+        _validation: function () {
+            var that = this,
+                element = that.element;
+
+            that._validationIcon = $("<span>")
+                .addClass("k-icon k-i-warning")
```
Comment (Stamo-Gochev) [inline src/kendo.numerictextbox.js]: Just and idea - you can extract the names of the classes into separate variables (constants), so they can be reused. It is also preferable to use the self-closing version of the tag, i.e. `$("<span />")`.

An alternative is to use something like:
```
var VALIDATION_ICON = "<span class='k-icon k-i-warning></span>";

that._validationIcon = $(VALIDATION_ICON).insertAfter()...
```
but this depends whether or not `k-icon` and `k-i-warning` are used in more than one places (the same applies to the icon rendering).

File: src/kendo.numerictextbox.js [hunk 2]
```diff
@@ -501,12 +513,27 @@ var __meta__ = { // jshint ignore:line
...
         },
 
+        _blinkInvalidState: function () {
+            var that = this,
+                wrapper = that._inputWrapper.addClass(INVALID);
+
+            that._validationIcon.show();
+            clearTimeout(that._blinkTimeout);
+
+            that._blinkTimeout = setTimeout(function(){
```
Comment (Stamo-Gochev) [inline src/kendo.numerictextbox.js]: The code inside the timeout function can be extracted to a separate method, such as `_removeInvalidState` (make sure to call it with `$.proxy` to be sure of the this object).

File: tests/textbox/api.js [hunk 3]
```diff
@@ -505,4 +505,25 @@
 
         equal(textbox._text.attr("title"), 20);
     });
+
+    asyncTest("input blinks when invalid input is entered.", function () {
```
Comment (Stamo-Gochev) [inline tests/textbox/api.js]: The test can be split in 3 new tests, which test for:
1. The input doesn't have an invalid decoration (the first 2 assertions).
2. The keypress event adds invalid decoration (the next 2 assertions).
3. A keypress event is triggered and after a certain time, the invalid decoration is removed (the last 2 assertions).

You can also think about adding a new test file, e.g. `validation.js`, which seems to be more appropriate for such tests in comparison to `api.js`.

File: src/kendo.numerictextbox.js (hunk 2 above)
Comment (dbalinov) [inline src/kendo.numerictextbox.js]: Adding 
`var setTimeout = window.setTimeout;`
breaks the tests
kendo.ui.NumericTextBox API focusing text input should call _focus FAILED

**Comment by dbalinov** [comment]
All comments from @Stamo-Gochev  are fixed except the one for `setTimeout`, as described in the [previous comment](https://github.com/telerik/kendo-ui-core/pull/2919#discussion_r106681320).

File: src/kendo.numerictextbox.js (hunk 2 above)
Comment (Stamo-Gochev) [inline src/kendo.numerictextbox.js]: Can you re-check why this is happening? Maybe you need to ensure that the `this` object is the correct one. There should be no difference between `setTimeout` and `window.setTimeout` as calling `setTimeout` will search for such a method in the scope chain and actually end in `window.setTimeout` as there is no "local" method, which overrides it.

In addition, run the tests and make sure that the focus stays in the browser that runs the tests, as changing the focused element might be the problem.

**Comment by Stamo-Gochev** [comment]
The CI builds are failing - can you re-check why this is happening?

**Comment by Stamo-Gochev** [comment]
Looks good, you can squash the 2 commits and merge them to master.

**Pull request merged**